### PR TITLE
Add documentation for tasks

### PR DESCRIPTION
**Pull request by Jdww**
https://teamleader.atlassian.net/browse/CAL-1053

Adding documentation for the tasks api.

### CODE REVIEW

**Comment by Jdww** [comment]
@stivni @jenssegers We are not sure under what category we have to place tasks. It doesn't really belong under calendar. Any ideas?

File: src/apiary.apib [hunk 1]
```diff
@@ -43,6 +43,8 @@ HOST: https://api.teamleader.eu
 
 :[Events](./04-calendar/events.apib)
 
+:[Tasks](./04-calendar/tasks.apib)
```
Comment (psrpinto) [inline src/apiary.apib]: PP: Remove this for now so we can merge this PR

**Review by psrpinto — DISMISSED** [review]
I would keep it under Calendar for now, we can move it later if needed.

**Review by r1dotl — DISMISSED** [review]

**Comment by psrpinto** [comment]
We decided to move it to the root level, so there will now be a Tasks section.

**Review by rathesDot — DISMISSED** [review]

**Review by psrpinto — DISMISSED** [review]

**Review by r1dotl — DISMISSED** [review]

**Review by rathesDot — DISMISSED** [review]

**Comment by psrpinto** [comment]
DO NOT MERGE since code isn't deployed yet

**Review by psrpinto — DISMISSED** [review]

**Review by r1dotl — DISMISSED** [review]

File: src/08-tasks/tasks.apib [hunk 2]
```diff
@@ -59,7 +59,10 @@ Create a new task.
 
     + Attributes (object)
         + description (string, required)
-        + due_at: `2016-02-04T16:44:33+00:00` (string, optional)
+        + due_at: `2016-02-04T16:44:33+00:00` (string, required)
+        + work_type_id: `32665afd-1818-0ed3-9e18-a603a3a21b95` (string, required)
+        + estimated_duration: 3600 (number, optional)
```
Comment (adognameddog) [inline src/08-tasks/tasks.apib]: Is it clear enough that this is in seconds? 3600 is a big hint of course

Comment (psrpinto) [reply src/08-tasks/tasks.apib]: Damn, we did that in #294 but forgot to do it here. Fixing

**Review by psrpinto — DISMISSED** [review]

File: apiary.apib [hunk 3]
```diff
@@ -1920,6 +1920,112 @@ Cancel a calendar event (for all attendees).
...
+                    + field: `due_at`
+                    + order: `asc`
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (array)
+            + (object)
+                + id: `f29abf48-337d-44b4-aad4-585f5277a456` (string)
+                + description (string)
```
Comment (tiagobastos) [inline apiary.apib]: PP: Add a description example
Same on the other requests/responses.

**Review by WouterSioen — DISMISSED** [review]
Even better would've probably been to make the estimated_duration some kind of value object containing a value and a unit, even when you only allow one unit. This might make it more future proof (we're doing the same thing with driving distance in work orders now)

File: apiary.apib [hunk 4]
```diff
@@ -2937,6 +2937,94 @@ Update a milestone.
...
+
+Create a new task.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + description (string, required)
+        + due_at: `2016-02-04T16:44:33+00:00` (string, required)
+        + work_type_id: `32665afd-1818-0ed3-9e18-a603a3a21b95` (string, required)
+        + estimated_duration: 3600 (number, optional)
```
Comment (stivni) [inline apiary.apib]: seconds?

**Review by tiagobastos — DISMISSED** [review]

File: apiary.apib (hunk 4 above)
Comment (stivni) [reply apiary.apib]: Q: this field can be written, but not read?

**Review by r1dotl — DISMISSED** [review]

File: apiary.apib (hunk 4 above)
Comment (psrpinto) [reply apiary.apib]: That's coming in other PRs. This PR documents what had already been implemented, since a while, as the Todos API.

**Review by rathesDot — DISMISSED** [review]

**Comment by Jdww** [comment]
We agree it makes sense to add a duration vo, we'll do that in a follow up pr. We've removed tasks from the apiary so this can be merged without actually releasing our task documentation. There are a few other pull requests open already that are depending on this pull request and waiting for the duration implementation will make everything a mess.

**Review by psrpinto — APPROVED** [review]

**Review by tiagobastos — APPROVED** [review]

**Review by r1dotl — APPROVED** [review]

**Pull request merged**